quic/client/handshake/FizzClientHandshakeContext.h

```diff
@@ -0,0 +1,25 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ *
+ */
+
+#pragma once
+
+#include <quic/client/handshake/ClientHandshakeFactory.h>
+
+namespace quic {
+
+class FizzClientHandshake;
+
+class FizzClientHandshakeContext
+    : public ClientHandshakeFactory,
+      public std::enable_shared_from_this<FizzClientHandshakeContext> {
+ public:
+  std::unique_ptr<ClientHandshake> makeClientHandshake(
```

Comment:
maybe add a comment here saying that the ClientHandshake might refer to the QuicCryptoState, so the lifetime of the QuicCryptoState should be longer than the returned ClientHandshake object.